azkaban-exec-server/src/main/java/azkaban/container/FlowContainer.java

```diff
@@ -0,0 +1,336 @@
+      } else {
+        logger.error("Failed to delete the cert file");
+        throw new IOException("Failed to delete the cert file");
+      }
+      if (keyStore == null) {
+        logger.error("Failed to Prefetch KeyStore");
+        throw new IOException("Failed to Prefetch KeyStore");
+      }
+    }
+  }
+
+  public static void main(final String[] args) throws IOException, ExecutorManagerException {
+    // Redirect all std out and err messages into log4j
+    StdOutErrRedirect.redirectOutAndErrToLog();
+    // Get all the arguments
+    final String execIdStr = args[0];
+    final String projectZipName = args[1];
+    // Process Execution ID.
+    int execId = 0;
+    try {
+      execId = Integer.parseInt(execIdStr);
+    } catch (NumberFormatException ne) {
+      System.out.printf("Execution ID %s is invalid %n", args[0]);
+    }
+    System.out.printf("Execution ID = %d%n", execId);
+
+    // Setup work directories
+    final Path currentWorkingDir = Paths.get("").toAbsolutePath();
+    final Path projectDirPath = Paths.get(currentWorkingDir.toString(), PROJECT_DIR);
+    final Path jobtypePluginPath = Paths.get(currentWorkingDir.toString(), JOBTYPE_DIR);
+
+    // Move files to respective dirs
+    // Create project dir
+    // TODO: Need to revisit
+    System.out.println("Creating project dir");
+    Files.createDirectory(projectDirPath);
+    Path projectZipPath = Paths.get(projectDirPath.toString(), projectZipName);
+    System.out.println("moving projectDir:" + projectZipName + ": to " +
+        projectZipPath);
+    Files.move(Paths.get(projectZipName), projectZipPath);
+    // TODO : revisit this logic,
+    // Unzip the project zip
+    FlowContainer.unzipFile(projectZipPath.toString(), projectDirPath);
+
+    // TODO : Deepak - throw away after directory setup is final.
+    Path pwd = Paths.get("").toAbsolutePath();
+    System.out.println("Deepak : Dumping dir structure for = " + pwd.toString());
+    java.nio.file.Files.walk(pwd).filter(java.nio.file.Files::isRegularFile)
+        .forEach(System.out::println);
+
+    // Set Azkaban props
+    Props props = setAzkabanProps(jobtypePluginPath);
+
+    // Inject AzkabanCommonModule
+    final Injector injector = Guice.createInjector(
+        new AzkabanCommonModule(props)
+    );
+    SERVICE_PROVIDER.setInjector(injector);
+
+    // Constructor
+    final FlowContainer flowContainer =
+        new FlowContainer(projectDirPath, props);
+
+    // Use some execId to execute the flow
+    flowContainer.submitFlow(execId);
+  }
+
+  // Set Azkaban Props
+  private static Props setAzkabanProps(final Path jobtypePluginPath) {
+    final Map<String, String> propsMap = new HashMap<>();
+    propsMap.put(AzkabanExecutorServer.JOBTYPE_PLUGIN_DIR,
+        jobtypePluginPath.toString());
+
+    // Setup the azkaban.properties here.
+    final String[] args = {CONF_ARG, CONF_DIR};
+    final Props props = AzkabanServer.loadProps(args);
+
+    // Set db stuff.
```

Comment:
What is this? Please write proper comment here.

Reply:
Old stale comment, removed.